src/d3a/d3a_core/simulation.py

```diff
@@ -363,13 +371,19 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
 
                 self.area.tick_and_dispatch()
                 self.area.update_area_current_tick()
+                if self.simulation_config.external_connection_enabled and \
+                        self.external_tick_counter.is_it_time_for_external_tick(tick_no):
+                    self.global_objects.update()
 
                 self.simulation_config.external_redis_communicator.\
                     publish_aggregator_commands_responses_events()
 
                 self.handle_slowdown_and_realtime(tick_no)
                 self.tick_time_counter = time()
 
+            if self.simulation_config.external_connection_enabled:
+                self.external_tick_counter.reset()
```

Comment:
As a followup from the comment above, if the GlobalStatistics class includes the external_tick_counter, then this reset() method could be populated automatically by the update() method (by adding one extra parameter which will indicate when does this update take place, on a tick event or a market event )